Compiling `namada_sdk` crate failed due to no "rand"

When building an application importing `namada_sdk` with `v0.26.0`,
```
error[E0599]: no function or associated item named `generate` found for struct `namada_core::types::key::ed25519::SigScheme` in the current scope
   --> /Users/yuji/.cargo/git/checkouts/namada-d9874c03bc334f5d/444da00/sdk/src/wallet/mod.rs:235:52
    |
235 |         SchemeType::Ed25519 => ed25519::SigScheme::generate(csprng).try_to_sk(),
    |                                                    ^^^^^^^^ function or associated item not found in `SigScheme`

error[E0599]: no function or associated item named `generate` found for struct `namada_core::types::key::secp256k1::SigScheme` in the current scope
   --> /Users/yuji/.cargo/git/checkouts/namada-d9874c03bc334f5d/444da00/sdk/src/wallet/mod.rs:237:35
    |
237 |             secp256k1::SigScheme::generate(csprng).try_to_sk()
    |                                   ^^^^^^^^ function or associated item not found in `SigScheme`

error[E0599]: no function or associated item named `generate` found for struct `namada_core::types::key::common::SigScheme` in the current scope
   --> /Users/yuji/.cargo/git/checkouts/namada-d9874c03bc334f5d/444da00/sdk/src/wallet/mod.rs:239:50
    |
239 |         SchemeType::Common => common::SigScheme::generate(csprng).try_to_sk(),
    |                                                  ^^^^^^^^ function or associated item not found in `SigScheme`

For more information about this error, try `rustc --explain E0599`.
error: could not compile `namada_sdk` (lib) due to 3 previous errors
```

`generate` requires `rand` in core.
https://github.com/anoma/namada/blob/e89a5a9214d40d367ee000699c98748544aa3d00/core/src/types/key/mod.rs#L314-L318

But `rand` isn't specified.
https://github.com/anoma/namada/blob/444da002513c60501fbcabab52ac2e8744b07b04/sdk/Cargo.toml#L71

--- a/.changelog/unreleased/bug-fixes/2164-fix-sdk-feature-flag.md
+++ b/.changelog/unreleased/bug-fixes/2164-fix-sdk-feature-flag.md
@@ -1,0 +1,2 @@
+- Fix a feature flag to compile namada_sdk
+  ([\#2164](https://github.com/anoma/namada/issues/2164))

--- a/sdk/Cargo.toml
+++ b/sdk/Cargo.toml
@@ -15,7 +15,7 @@ version.workspace = true
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [features]
-default = []
+default = ["tendermint-rpc"]
 
 multicore = ["masp_proofs/multicore"]
 

--- a/sdk/Cargo.toml
+++ b/sdk/Cargo.toml
@@ -68,7 +68,7 @@ futures.workspace = true
 itertools.workspace = true
 masp_primitives.workspace = true
 masp_proofs = { workspace = true, features = ["download-params"] }
-namada_core = {path = "../core", default-features = false}
+namada_core = {path = "../core", default-features = false, features = ["rand"]}
 namada_ethereum_bridge = {path = "../ethereum_bridge", default-features = false}
 namada_proof_of_stake = {path = "../proof_of_stake", default-features = false}
 num256.workspace = true

